### PR TITLE
Expose VisionDataSet class so it can be modified

### PR DESCRIPTION
**Pull request by bbowles-personal**
This is really helpful class, but I had a use case where I needed to modify slightly.

### CODE REVIEW

**Comment by pmeier** [comment]
@bbowles-personal I'm curious: What did you need to modify? Is this something we could also use? On what part was `VisionDataset` not general enough?

File: torchvision/datasets/__init__.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 from .caltech import Caltech101, Caltech256
 from .celeba import CelebA
 from .sbd import SBDataset
+from .vision import VisionDataset
 
 __all__ = ('LSUN', 'LSUNClass',
```
Comment (pmeier) [inline torchvision/datasets/__init__.py]: Shouldn't `VisionDataset` contained in `__all__` if we want to expose it?

Comment (fmassa) [reply torchvision/datasets/__init__.py]: If we want it to be imported via `from torchvision.datasets import *`, then yes.

Comment (pmeier) [reply torchvision/datasets/__init__.py]: Let me rephrase: Do we want that? An _all or nothing_ approach seems reasonable to me.

Comment (fmassa) [reply torchvision/datasets/__init__.py]: I'm fine with either, but for consistency let's add it there.

Comment (bbowles-personal) [reply torchvision/datasets/__init__.py]: In my case, I wanted to restrict the folders that are found by mucking with `_find_classes` but the easiest way to do that was just to inherit from VisionDataset and write my own `DatasetFolder` class.
If you want, I can write a pull to allow a keyword arg to ImageFolder to only allow it to find certain folders, which is actually what I need....
not really using this code for what it was intended to do so not sure if you really want that.
thanks for approving

Comment (fmassa) [reply torchvision/datasets/__init__.py]: I can merge this as is, but I'd accept a PR adding `VisionDataset` to `__all__`
